src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java

```diff
@@ -147,17 +147,24 @@ protected AbstractJDBCTestElement() {
     /**
      * Execute the test element.
      * 
-     * @param conn a {@link SampleResult} in case the test should sample; <code>null</code> if only execution is requested
+     * @param conn a {@link Connection}
      * @return the result of the execute command
      * @throws SQLException if a database error occurs
      * @throws UnsupportedEncodingException when the result can not be converted to the required charset
      * @throws IOException when I/O error occurs
      * @throws UnsupportedOperationException if the user provided incorrect query type 
      */
     protected byte[] execute(Connection conn) throws SQLException, IOException, UnsupportedOperationException {
+        return execute(conn,  new SampleResult());
+    }
+
+    /**
+     * Use the sample given as argument to set time to first byte in the "latency" field of the SampleResult.
+     */
+    protected byte[] execute(Connection conn, SampleResult sample) throws SQLException, IOException {
```

Comment:
The javadoc could (should) have documentation for the params, return value, ...
It looks a bit strange, that execute(conn) throws UnsupportedOperationException, but this method does not.